integrationTests/chainSimulator/common.go

```diff
@@ -366,6 +366,7 @@ func SetEsdtInWallet(
 func RegisterAndSetAllRoles(
 	t *testing.T,
 	cs ChainSimulator,
+	nodeHandler process.NodeHandler,
```

Comment:
I think we could call this "metaNodeHandler"

Reply:
Actually, I think a "smarter" way to reuse this would be to use `cs.GetShardID(vm.ESDTSCAddress)`.
What do you think about this approach?

Reply:
This way you also get rid of passing `nodeHandler`

Reply:
Implemented the node handler from esdt sc shard id